src/main/java/com/urswolfer/intellij/plugin/gerrit/git/GerritGitUtil.java

```diff
@@ -426,6 +427,15 @@ public GitLineHandler compute() {
         }
     }
 
+    public Optional<VcsFullCommitDetails> loadGitCommitInfo(Project project, GitRepository gitRepository, ChangeInfo changeInfo) throws VcsException {
+        List<GitCommit> history = GitHistoryUtils.history(project, gitRepository.getRoot(), changeInfo.id);
+        if(history.isEmpty()){
+            return Optional.absent();
+        }
+        // I expect there to be only 1 commit, since that's all we requested from the git history
```

Comment:
You could throw if there is more than one element in the list if you assume so?

Reply:
i have a good feeling about this.